hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -123,6 +126,20 @@ private static HoodieCommitMetadata buildMetadataFromStats(List<HoodieWriteStat>
     return commitMetadata;
   }
 
+  public static Option<HoodieCommitMetadata> buildMetadataFromInstant(HoodieDefaultTimeline timeline, HoodieInstant instant) {
```

Comment:
probably better to inline this method. Just do below wherever needed:
```
HoodieCommitMetadata commitMetadata = HoodieCommitMetadata.fromBytes(completedInstants.getInstantDetails(instant).get(), HoodieCommitMetadata.class);
```

Reply:
Retained the function (mainly to hide error handling)